src/vs/editor/contrib/linesOperations/test/common/linesOperations.test.ts

```diff
@@ -67,4 +67,118 @@ suite('Editor Contrib - Line Operations', () => {
 				assert.equal(model.getLineContent(5), 'horlworld', '005');
 			});
 	});
+
+	test('Join lines', function () {
+		withMockCodeEditor(
+			[
+				'hello',
+				'world',
+				'hello ',
+				'world',
+				'hello		',
+				'	world',
+				'hello   ',
+				'	world',
+				'',
+				'',
+				'hello world'
+			], {}, (editor, cursor) => {
+				let model = editor.getModel();
+				let joinLinesAction = new JoinLinesAction();
+
+				editor.setSelection(new Selection(1, 2, 1, 2));
+				joinLinesAction.run(null, editor);
+				assert.equal(model.getLineContent(1), 'hello world', '001');
+				assert.deepEqual(editor.getSelection().toString(), new Selection(1, 6, 1, 6).toString(), '002');
+
+				editor.setSelection(new Selection(2, 2, 2, 2));
+				joinLinesAction.run(null, editor);
+				assert.equal(model.getLineContent(2), 'hello world', '003');
+				assert.deepEqual(editor.getSelection().toString(), new Selection(2, 7, 2, 7).toString(), '004');
+
+				editor.setSelection(new Selection(3, 2, 3, 2));
+				joinLinesAction.run(null, editor);
+				assert.equal(model.getLineContent(3), 'hello world', '005');
+				assert.deepEqual(editor.getSelection().toString(), new Selection(3, 7, 3, 7).toString(), '006');
+
+				editor.setSelection(new Selection(4, 2, 5, 3));
+				joinLinesAction.run(null, editor);
+				assert.equal(model.getLineContent(4), 'hello world', '007');
+				assert.deepEqual(editor.getSelection().toString(), new Selection(4, 2, 4, 8).toString(), '008');
+
+				editor.setSelection(new Selection(5, 1, 7, 3));
+				joinLinesAction.run(null, editor);
+				assert.equal(model.getLineContent(5), 'hello world', '009');
+				assert.deepEqual(editor.getSelection().toString(), new Selection(5, 1, 5, 3).toString(), '010');
+			});
+	});
+
+	test('transpose', function () {
```

Comment:
does it work for empty lines, or on empty files?

You can use scripts\test.bat --coverage and my awesome LCOV plugin to see your code coverage ;)